File: ruby/URI_1040.rb
notas = gets.split

n1 = notas[0].to_f
n2 = notas[1].to_f
n3 = notas[2].to_f
n4 = notas[3].to_f

media = (n1*2+n2*3+n3*4+n4*1)/10.0
puts "Media: %.1f" % media

if (media >= 7) then
	puts "Aluno aprovado."
elsif (media < 5) then
	puts "Aluno reprovado."
else
	puts "Aluno em exame."

	exame = gets.to_f
	media = (media + exame) / 2.0

	puts "Nota do exame: %.1f" % exame

	if (media >= 5) then
		puts "Aluno aprovado."
	else
		puts "Aluno reprovado."
	end

	puts "Media final: %.1f" % media
end

File: c-plus-plus/URI_1040.cpp
#include <iostream>
#include <iomanip>
using namespace std;
int main () {
	double note1, note2, note3, note4, noteExtra;
	cin >> note1 >> note2 >> note3 >> note4;
	double average;
	average = ((note1 * 2) + (note2 * 3) + (note3 * 4) + (note4 * 1)) / 10.0;
	cout << "Media: " << fixed << setprecision(1) << average << endl;
	if (average >= 7.0)
	{
		cout << "Aluno aprovado." << endl;
	}
	else
	{
		if (average < 5.0)
		{
			cout << "Aluno reprovado." << endl;
		}
		else
		{
			cout << "Aluno em exame." << endl;
			cin >> noteExtra;
			cout << "Nota do exame: " << noteExtra << endl;
			average = (average + noteExtra) / 2.0;
			if (average >= 5.0)
			{
				cout << "Aluno aprovado." << endl;
			}
			else
			{
				cout << "Aluno reprovado." << endl;
			}
			cout << "Media final: " << fixed << setprecision(1) << average << endl;
		}
	}
	return 0;
}
